incubator/monochart/templates/daemonset.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ include "common.fullname" . }}
+  annotations:
+{{ toYaml .Values.daemonset.annotations | indent 4 }}
+  labels:
+{{ include "common.labels.standard" . | indent 4 }}
+spec:
+  template:
+    metadata:
+      name: {{ include "common.fullname" . }}
+      annotations:
+{{ toYaml .Values.daemonset.pod.annotations | indent 8 }}
+      labels:
+        app: {{ template "common.name" . }}
+        release: {{ .Release.Name | quote }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        imagePullSecrets:
+{{ toYaml .Values.image.pullSecrets | indent 10 }}
+        envFrom:
+        - configMapRef:
+            name: {{ include "common.fullname" . }}
+        env:
+{{- range $name, $value := .Values.env.open }}
+{{- if not (empty $value) }}
+        - name: {{ $name | quote }}
+          value: {{ $value | quote }}
+{{- end }}
+{{- end }}
+
+{{- $config_name := include "common.fullname" . }}
+{{- range $name, $value := .Values.env.open }}
+{{- if not ( empty $value) }}
+        - name: {{ $name | quote }}
+          valueFrom:
+            configMapKeyRef:
+              name: {{ $config_name }}
+              key: {{ $name | quote }}
+{{- end }}
+{{- end }}
+
+{{- $secret_name := include "common.fullname" . }}
+{{- range $name, $value := .Values.env.secret }}
+{{- if not ( empty $value) }}
+        - name: {{ $name | quote }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ $secret_name }}
+              key: {{ $name | quote }}
+{{- end }}
+{{- end }}
+
+        args: {}
+        ports:
+        - containerPort: {{ .Values.service.internalPort }}
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: {{ .Values.service.internalPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        readinessProbe:
+          httpGet:
+            path: /health
+            port: {{ .Values.service.internalPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        volumeMounts:
+        - mountPath: /data
+          name: storage
+        - mountPath: /config
+          name: config
+        - mountPath: /secret
+          name: secret
+          readOnly: true
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+      volumes:
+      - name: storage
+      {{- if .Values.persistence.Enabled }}
+        persistentVolumeClaim:
+          claimName: {{ .Values.persistence.ExistingClaim | default (include "common.fullname" .) }}
+      {{- else }}
+        emptyDir: {}
+      {{- end -}}
+
+      - name: config
```

Comment:
We need to generalize this configuration and give it a proper section in `values.yaml`:

e.g.
```
configFiles:
  "/etc/test.conf": |-
    [default]
    debug=true
```